tests/unittests/nominal/test_pearson.py

```diff
@@ -0,0 +1,133 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import itertools
+import operator
+from collections import namedtuple
+
+import pandas as pd
+import pytest
+import torch
+from scipy.stats.contingency import association
+
+from torchmetrics.functional.nominal.pearson import (
+    pearsons_contingency_coefficient,
+    pearsons_contingency_coefficient_matrix,
+)
+from torchmetrics.nominal.pearson import PearsonsContingencyCoefficient
+from torchmetrics.utilities.imports import _compare_version
+from unittests.helpers.testers import BATCH_SIZE, NUM_BATCHES, MetricTester
+
+Input = namedtuple("Input", ["preds", "target"])
+NUM_CLASSES = 4
+
+_input_default = Input(
+    preds=torch.randint(high=NUM_CLASSES, size=(NUM_BATCHES, BATCH_SIZE)),
+    target=torch.randint(high=NUM_CLASSES, size=(NUM_BATCHES, BATCH_SIZE)),
+)
+
+_input_logits = Input(
+    preds=torch.rand(NUM_BATCHES, BATCH_SIZE, NUM_CLASSES), target=torch.rand(NUM_BATCHES, BATCH_SIZE, NUM_CLASSES)
+)
+
+# No testing with replacing NaN's values is done as not supported in SciPy
+
+
+@pytest.fixture
+def _matrix_input():
+    matrix = torch.cat(
+        [
+            torch.randint(high=NUM_CLASSES, size=(NUM_BATCHES * BATCH_SIZE, 1), dtype=torch.float),
+            torch.randint(high=NUM_CLASSES + 2, size=(NUM_BATCHES * BATCH_SIZE, 1), dtype=torch.float),
+            torch.randint(high=2, size=(NUM_BATCHES * BATCH_SIZE, 1), dtype=torch.float),
+        ],
+        dim=-1,
+    )
+    return matrix
+
+
+def _sk_pearsons_t(preds, target):
```

Comment:
```suggestion
def _pd_pearsons_t(preds, target):
```
as sk used to be for SciKitLearn